nncf/quantization/api/engine.py

```diff
@@ -31,9 +29,7 @@ class Engine(ABC):
 
     def __init__(self):
         self.model = None
-        self._sampler = None
         self._dataset = None
```

Comment:
Do we really need a dataset here?

Reply:
It still uses in the `compute_statistics` method. I think we will remove it during `Engine` refactoring as the next step.